admin/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+		SubCommand::Stress(bench) => {
+
+			let args = bench.get_args();
+			// avoid deleting folders by mistake.
+			options.path.push("test_db_stress");
+			if options.path.exists() && !args.append {
+				std::fs::remove_dir_all(options.path.as_path())
+					.map_err(|e| format!("Error clearing stress db: {:?}", e))?;
+			}
+
+			use crate::bench::BenchDb;
+			let db = bench::BenchAdapter::with_options(&options);
+
+			crate::bench::run_internal(args, db);
+		},
+	}
+	Ok(())
+}
+
+/// Admin cli command for parity-db.
+#[derive(Debug, StructOpt)]
+pub struct Shared {
+	/// Specify db base path.
+	#[structopt(long, short = "d", value_name = "PATH", parse(from_os_str))]
+	pub base_path: Option<PathBuf>,
+
+	/// Do not sync file on each flush.
+	#[structopt(long)]
+	pub no_sync: bool,
+
+	/// Register stat from those admin operations.
+	#[structopt(long)]
+	pub with_stats: bool,
+
+	/// Indicate the number of column, when using
+	/// a new or temporary db, defaults to one.
+	#[structopt(long)]
+	pub nb_columns: Option<u8>,
+
+	/// Indicate a metadata configuration to use.
+	#[structopt(long)]
+	pub use_metadata: Option<PathBuf>,
+
+	/// Sets a custom logging filter. Syntax is <target>=<level>, e.g. -lsync=debug.
+	///
+	/// Log levels (least to most verbose) are error, warn, info, debug, and trace.
+	/// By default, all targets log `info`. The global log level can be set with -l<level>.
+	#[structopt(short = "l", long, value_name = "LOG_PATTERN")]
+	pub log: Vec<String>,
+}
+
+/// Admin cli command for parity-db.
+#[derive(Debug, StructOpt)]
+pub struct Cli {
+
+	/// Subcommands.
+	#[structopt(subcommand)]
+	pub subcommand: SubCommand,
+
+	/// Enable validator mode.
+	///
+	/// The node will be started with the authority role and actively
+	/// participate in any consensus task that it can (e.g. depending on
+	/// availability of local keys).
+	#[structopt(
+		long = "validator"
+	)]
+	pub validator: bool,
+}
+
+#[derive(Debug, StructOpt)]
+pub enum SubCommand {
+	/// Show stats.
+	Stats(Stats),
+	/// Migrate db (update version or change collection options).
+	Migrate(Migrate),
+	/// Run db until all logs are flushed.
+	Flush(Flush),
+	/// Check db content.
+	Check(Check),
+	/// Stress tests.
+	Stress(bench::Stress),
+}
+
+impl Cli {
+	fn shared(&self) -> &Shared {
+		match &self.subcommand {
+			SubCommand::Stats(stats) => {
+				&stats.shared
+			},
+			SubCommand::Migrate(stats) => {
+				&stats.shared
+			},
+			SubCommand::Flush(flush) => {
+				&flush.shared
+			},
+			SubCommand::Check(check) => {
+				&check.shared
+			},
+			SubCommand::Stress(bench) => {
+				&bench.shared
+			},
+		}
+	}
+}
+
+/// Show stats for columns.
+#[derive(Debug, StructOpt)]
+pub struct Stats {
+	#[structopt(flatten)]
+	pub shared: Shared,
+
+	/// Only show stat for the given column.
+	#[structopt(long)]
+	pub column: Option<u8>,
+
+	/// Clear current stats.
+	#[structopt(long)]
+	pub clear: bool,
+}
+
+/// Migrate db (update version or change collection options).
```

Comment:
Using multiple terms for the same thing is a source of confusion. Let's stick to "column"